### PR TITLE
Fix Data browser shows the same data in all views

### PR DESCRIPTION
**Pull request by Karakatiza666**
Fix https://github.com/feldera/feldera/issues/993

### CODE REVIEW

File: web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts [hunk 1]
```diff
@@ -29,15 +29,33 @@ const updateRowsIntegral = (oldRows: Map<number, Row>, deltaRows: Row[]) => {
...
+    setRows(old => {
+      const relationChanged = !old.length
+      if (relationChanged) {
+        integral.clear()
+      }
+      updateRowsIntegral(integral, parsedRows)
+      if (!relationChanged) {
+        return old
+      }
+      return Array.from(integral.values()).sort((a, b) => a.genId - b.genId)
```
Comment (gz) [inline web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts:55]: It's not clear to me why it needs to be sorted? (comment would help)

**Review by gz — APPROVED** [review]

File: web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts [hunk 2]
```diff
@@ -29,15 +29,33 @@ const updateRowsIntegral = (oldRows: Map<number, Row>, deltaRows: Row[]) => {
   return oldRows
 }
 
-const computeAndSet = (prev: Map<number, Row>) => (setRows: Dispatch<SetStateAction<Row[]>>, parsedRows: Row[]) => {
-  updateRowsIntegral(prev, parsedRows)
-  setRows(Array.from(prev.values()).sort((a, b) => a.genId - b.genId))
-  return prev
+const computeAndSet = (integral: Map<number, Row>) => (setRows: Dispatch<SetStateAction<Row[]>>, parsedRows: Row[]) => {
+  setRows(old => {
+    const relationChanged = !old.length
```
Comment (gz) [inline web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts:34]: is this sufficient to indicate relation has changed? no false positives?

File: web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts (hunk 1 above)
Comment (Karakatiza666) [reply web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts:55]: I am not sure either - it is the way it was initially, see the initial code in https://github.com/feldera/feldera/commit/b4038cccea968f4c53f59e1f059a78b5db4c40eb , line 90

File: web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts (hunk 2 above)
Comment (Karakatiza666) [reply web-console/src/lib/compositions/streaming/inspection/useTableUpdater.ts:34]: The only false positive would be when on update relation didn't change but it was empty, in which case it's not a problem as the behavior of update is idempotent

**Comment by Karakatiza666** [comment]
Also fixes https://github.com/feldera/cloud/issues/101 and https://github.com/feldera/feldera/issues/999

**Pull request merged**